web_tool_script_1.R

```diff
@@ -9,11 +9,14 @@ source("0_web_functions.R")
 source("0_json_functions.R")
 source("0_portfolio_test.R")
 
+if (!exists("portfolio_name_ref_all")) { portfolio_name_ref_all <- "TestPortfolio_Input" }
+if (!exists("portfolio_root_dir")) { portfolio_root_dir <- "working_dir" }
+
 setup_project()
```

Comment:
I don't remember what `setup_project()` does but its name seems vague enough to fit the assignment of these two variables, and maybe `set_webtool_paths()` too. That way you could remove duplication in all three web tool scripts.

Reply:
I don't know what `setup_project()` does either... I thought that was something you wrote... hahahaha

Reply:
Yes, I think I wrote it but I right now I didn't bother to look into it to see what it does. But if I named it properly, then it's a catch-all interface to remove duplication like this.